### PR TITLE
update `deployment_crud.py` to include a `restart_deployment` method for typed client

### PR DESCRIPTION
**Pull request by Priyankasaggu11929**
#### What type of PR is this?
/kind documentation
/kind cleanup

#### What this PR does / why we need it:
This PR updates the `examples/deployment_crud.py` python script, to add a `rolling_restart_deployment` method. This new method demonstrates how to perform a rolling restart on a `deployment` resource using the k8s python `typed client`.

The output of the script looks like:
Screenshot from 2021-05-13 13-54-53

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes-client/python/issues/1378

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by yliaog** [comment]
thanks for the pr
/lgtm
/approve

**Comment by Priyankasaggu11929** [comment]
Hi @yliaog, some of the travis CI jobs are failing. I’m unable to understand the issues here. Could you please point me with what might be a fix for this?

https://travis-ci.org/github/kubernetes-client/python/jobs/769364787#L329

14E691B7-AEBB-4C7A-8A24-C1CD4807AAF9

thank you.

**Comment by Priyankasaggu11929** [comment]
Hi @yliaog, some of the travis CI jobs are failing. I’m unable to understand the issues here. Could you please point me with what might be wrong?

https://travis-ci.org/github/kubernetes-client/python/jobs/769364787#L329

14E691B7-AEBB-4C7A-8A24-C1CD4807AAF9

**Comment by yliaog** [comment]
from https://travis-ci.org/github/kubernetes-client/python/jobs/769364787#L329

/home/travis/build/kubernetes-client/python/kubernetes/../examples/deployment_crud.py:16:80: E501 line too long (84 > 79 characters)
/home/travis/build/kubernetes-client/python/kubernetes/../examples/deployment_crud.py:89:80: E501 line too long (81 > 79 characters)

**Comment by Priyankasaggu11929** [comment]
> from https://travis-ci.org/github/kubernetes-client/python/jobs/769364787#L329
> 
> /home/travis/build/kubernetes-client/python/kubernetes/../examples/deployment_crud.py:16:80: E501 line too long (84 > 79 characters)
> /home/travis/build/kubernetes-client/python/kubernetes/../examples/deployment_crud.py:89:80: E501 line too long (81 > 79 characters)

Thank you so much for pointing. I'm trying to fix them.

**Comment by yliaog** [comment]
closing to trigger CI
/close

**Comment by Priyankasaggu11929** [comment]
@yliaog, I'll re-write this script to make the code lines smaller. 

The `update-pycodestyle` script is still breaking failing the `travis-ci` jobs (I've tried formatting the code with a local `pycodestyle` module & then pushing the changes, but it didn't work with the failing jobs)

**Comment by yliaog** [comment]
diff --git a/examples/deployment_crud.py b/examples/deployment_crud.py
index 832e461..8563d9a 100644
--- a/examples/deployment_crud.py
+++ b/examples/deployment_crud.py
@@ -20,9 +20,10 @@ The example covers the following:
 """
 
 import datetime
+
 import pytz
-from kubernetes import client, config
 
+from kubernetes import client, config
 
 DEPLOYMENT_NAME = "nginx-deployment"

**Comment by roycaihw** [comment]
looks like there is still some codestyle test failure: https://travis-ci.org/github/kubernetes-client/python/builds/769827574

File: examples/deployment_crud.py [hunk 1]
```diff
@@ -13,96 +13,137 @@
...
+    print("\n[INFO] deployment `nginx-deployment` restarted.\n")
+    print("%s\t\t\t%s\t%s" % ("NAME", "REVISION", "RESTARTED-AT"))
+    print(
+        "%s\t%s\t\t%s\n"
+        % (
+            resp.metadata.name,
+            resp.metadata.generation,
+            resp.spec.template.metadata.annotations,
+        )
+    )
```
Comment (roycaihw) [inline examples/deployment_crud.py:135]: to fix the pycodestyle test failure, please remove the trailing spaces in this line.

Comment (Priyankasaggu11929) [reply examples/deployment_crud.py]: @roycaihw, I fixed the whitespaces issues but the job still remain failing.

Another thing to note:

Just to figure out what exactly amongst the new changes I added,  is a breaking change. I started by adding just the following  import commands (on top of the example file in master branch)

```
+ import datetime
+ import pytz
   from kubernetes import client, config
```

The job is still failing. https://travis-ci.org/github/kubernetes-client/python/jobs/770894221

I'm not sure if I'm missing something really simple here?

Comment (roycaihw) [reply examples/deployment_crud.py]: pycodestyle is asking for some empty lines between the imports. Sorry the tool has a strong opinion about standard v.s. third-party imports-- imports in different categories need to be separated by empty lines. 
```
diff --git a/examples/deployment_crud.py b/examples/deployment_crud.py
index 4fd02a7..7dcd6b3 100644
--- a/examples/deployment_crud.py
+++ b/examples/deployment_crud.py
@@ -17,7 +17,9 @@ Creates, updates, and deletes a deployment using AppsV1Api.
 """
 
 import datetime
+
 import pytz
+
 from kubernetes import client, config
 
 DEPLOYMENT_NAME = "nginx-deployment"
ERROR: InvocationError for command /home/travis/build/kubernetes-client/python/scripts/update-pycodestyle.sh (exited with code 1)
```

Comment (Priyankasaggu11929) [reply examples/deployment_crud.py]: @roycaihw, @yliaog, finally everything is green now :smile: 

Sorry, it caused a lot of trouble due to repeated failing tests (& me not understanding what exactly it required). But, I definitely learnt a lot about the tool.

Thank you so much.

**Pull request merged**